jaeger-core/src/main/java/com/uber/jaeger/utils/RateLimiter.java

```diff
@@ -22,20 +22,20 @@
 package com.uber.jaeger.utils;
 
 public class RateLimiter {
-  private final double creditsPerSecond;
   private final double creditsPerNanosecond;
   private final Clock clock;
   private double balance;
+  private double maxBalance;
   private long lastTick;
 
-  public RateLimiter(double creditsPerSecond) {
-    this(creditsPerSecond, new SystemClock());
+  public RateLimiter(double creditsPerSecond, double maxBalance) {
+    this(creditsPerSecond, maxBalance, new SystemClock());
   }
 
-  public RateLimiter(double creditsPerSecond, Clock clock) {
+  public RateLimiter(double creditsPerSecond, double maxBalance, Clock clock) {
```

Comment:
I think this interface should be something like `public RateLimiter(double rate, Seconds timePeriod)`
This reduces the cognitive overhead of the code and allows a reader to easily determine how many requests per a given time period is allowed by the RateLimiter. This is also consistent with the behavior of rate limiter interface before your commit. 

I don't like the interface you propose because it exposes the internals of a leaky bucket rate limiter, and expects users to be familiar with how leaky bucket rate limiting works (as opposed to saying give me a rate limiter that lets in `x` requests in `y` seconds)